src/Simplex_tree/include/gudhi/Simplex_tree/multi_filtrations/Finitely_critical_filtrations.h

```diff
@@ -0,0 +1,1051 @@
+    return multi_filtration_.end();
+  }
+
+
+  /*
+  * Same as its one critical counterpart.
+  * If a grid is given, projects multifiltration_ on this grid and returns
+  * multi critical filtration composed of the coordinates in the given grid
+  *
+  */
+  template <typename oned_array>
+  inline Multi_critical_filtration<std::int32_t>
+  coordinates_in_grid(const std::vector<oned_array>& grid) const {
+    assert(grid.size() >= this->num_generators());
+    Multi_critical_filtration<std::int32_t> out(this->num_generators());
+    for (std::size_t i = 0u; i < this->num_generators(); ++i) {
+      out[i] = this->multi_filtration_[i].coordinates_in_grid(grid);
+    }
+    return out;
+  }
+  /*
+  * Same as `coordinates_in_grid`, but does the operation in-place.
+  *
+  */
+  template <typename oned_array>
+  inline void coordinates_in_grid_inplace(const std::vector<oned_array>& grid,
+                                          bool coordinate = true) {
+    assert(grid.size() >= this->num_generators());
+    for (auto &x : this->multi_filtration_) {
+      x.coordinates_in_grid_inplace(grid, coordinate);
+    }
+  }
+  template <typename U> inline Multi_critical_filtration<U> astype() const {
+    std::vector<One_critical_filtration<U>> out(this->num_generators());
+    for (std::size_t i = 0u; i < this->num_generators(); ++i) {
+      out[i] = this->multi_filtration_[i].template astype<U>();
+    }
+    return Multi_critical_filtration<U>(std::move(out));
+  }
+  inline void push_back(const OneCritical& x){
+    multi_filtration_.push_back(x);
+  }
+  inline const std::vector<One_critical_filtration<T>>& as_vector() const {
+    return multi_filtration_;
+  }
+
+  inline std::vector<std::vector<T>> as_VECTOR() const {
+    std::vector<std::vector<T>> out(this->num_generators(),
+                                    std::vector<T>(this->num_parameters()));
+    for (std::size_t i = 0; i < this->num_generators(); ++i) {
+      for (std::size_t j = 0; j < this->num_parameters(); ++j) {
+        out[i][j] = multi_filtration_[i][j];
+      }
+    }
+    return out;
+  }
+
+  inline void _clean(bool keep_inf=true) {
+    multi_filtration_.erase(std::remove_if(multi_filtration_.begin(), multi_filtration_.end(),
+                              [keep_inf](const OneCritical &a) {
+                                return a.empty() ||
+                                       ((!keep_inf) &&
+                                        (a.is_inf() || a.is_minus_inf()));
+                              }),
+                            multi_filtration_.end());
+  }
+  inline std::size_t num_generators() const {
+    return multi_filtration_.size();
+  }
+
+
+
+  // TODO : this costs a lot... optimize / cheat in some way for python ?
+  /*
+  * Checks if `this`, seen as a birth curve is under the `other` birth curve,
+  *
+  */
+  inline bool operator<(const Multi_critical_filtration<T,co> &other) const {
+    //check if this curves is below other's curve 
```

Comment:
You say a < b if the curve of a is below the curve of b **and they don't touch anywhere**?
Usually, I would expect: the curve of a is below the curve of b and there is at least 1 place where they differ (they are not equal), although asking for the 2 curves not to touch at all could have some uses I guess. Can you shed some light on how this will be used?